src/libraries/System.Private.CoreLib/src/System/Decimal.DecCalc.cs

```diff
@@ -192,29 +192,70 @@ private static void UInt64x64To128(ulong a, ulong b, ref DecCalc result)
             /// <param name="bufNum">96-bit dividend as array of uints, least-sig first</param>
             /// <param name="den">32-bit divisor</param>
             /// <returns>Returns remainder. Quotient overwrites dividend.</returns>
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
             private static uint Div96By32(ref Buf12 bufNum, uint den)
             {
-                // TODO: https://github.com/dotnet/runtime/issues/5213
-                ulong tmp, div;
-                if (bufNum.U2 != 0)
+                if (X86.X86Base.X64.IsSupported)
```

Comment:
64bit hardware divide used to be slow on intel processors before Ice lake, however it seems to still be better than the current code in main.
Current hardware (Ryzen and Ice lake +) has best performance with 64bit divide

I ran  [some benchmarks including older hardware](https://github.com/Daniel-Svensson/ClrExperiments/blob/master/ClrDecimal/Benchmarks/results/Div96By32_benchmarks_2024.txt)  

It is just a small performance improvement on skylake with 64bit divide for worst case 19ns ((instead of 22ns) , in contrast with 32bit divide (<7ns)